code/game/objects/items/misc.dm

```diff
@@ -22,3 +23,79 @@
 	throw_speed = 1
 	throw_range = 20
 	flags = CONDUCT
+
+/obj/item/petcollar
+	name = "pet collar"
+	desc = "The latest fashion accessory for your favorite pets!"
+	icon_state = "petcollar"
+	item_color = "petcollar"
+	var/tagname = null
+	var/obj/item/card/id/access_id
+
+/obj/item/petcollar/Destroy()
+	QDEL_NULL(access_id)
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/petcollar/attack_self(mob/user as mob)
+	var/option = "Change Name"
+	if(access_id)
+		option = input(user, "What do you want to do?", "[src]", option) as null|anything in list("Change Name", "Remove ID")
+
+	switch(option)
+		if("Change Name")
+			var/t = input(user, "Would you like to change the name on the tag?", "Name your new pet", tagname ? tagname : "Spot") as null|text
+			if(t)
+				tagname = copytext(sanitize(t), 1, MAX_NAME_LEN)
+				name = "[initial(name)] - [tagname]"
+		if("Remove ID")
+			if(access_id)
+				user.visible_message("<span class='warning'>[user] starts unclipping \the [access_id] from \the [src].</span>")
```

Comment:
```suggestion
				user.visible_message("<span class='warning'>[user] starts unclipping [access_id] from [src].</span>")
```

Reply:
All suggestions applied and tested, thanks lewcc!